### ISSUE TITLE
[SOLVED] run clock.matrix app

### ISSUE THREAD

**Issue by nebiljabari** [matrix-io/matrix-os#10]
@eighteyes  Could you be more explicite about this line in the ReadMe :

> 3 - If you want to start a MATRIX app on launch, use the env START_APP. ex. START_APP=monitor node index.js

`monitor` stand for what ? My-app folder ? If yes, "where" exactly my-app should be. In `matrix-os/apps` ?

And how can I make run your clock.matrix app example ? I only receive dots in my terminal

**Comment by eighteyes** [comment]
Dots in the terminal means you are running the app. The dots represent the light.  Are you not seeing the LEDS change?

**Comment by nebiljabari** [comment]
Really want to say yes but no LEDs light up

**Comment by eighteyes** [comment]
Are you running malos in another terminal?

**Comment by nebiljabari** [comment]
Absolutly:

```
pi@raspberrypi:~ $ malos
**************
MALOS starting
**************

You can query specific driver info using port 20012.
Registered driver IMU with port 20013.
Registered driver Humidity with port 20017.
Registered driver Everloop with port 20021.
Registered driver Pressure with port 20025.
Registered driver UV with port 20029.
```

**Comment by nebiljabari** [comment]
Plus, when I enter this line : `START_APP=monitor node index.js` in `~/matrix-os` folder I get this error:

Local Apps: [ 'clock.matrix' ]
Local / Installed Δ 1
MXSS Connected: http://rc-mxss.admobilize.com
=<[^\/^]>= [my-device-id] ready
 _  _ ____ ___ ____ _ _  _ 
 |\/| |__|  |  |__/ |  \/   [o__o]
 |  | |  |  |  |  \ | _/\_  v0.5.0
Error: No user installed firebase application: monitor
    at /home/pi/matrix-os/node_modules/matrix-firebase/lib/app.js:189:19
    at /home/pi/matrix-os/node_modules/matrix-firebase/node_modules/firebase/database-node.js:144:614
    at Qb (/home/pi/matrix-os/node_modules/matrix-firebase/node_modules/firebase/database-node.js:43:165)
    at nc (/home/pi/matrix-os/node_modules/matrix-firebase/node_modules/firebase/database-node.js:31:216)
    at mc (/home/pi/matrix-os/node_modules/matrix-firebase/node_modules/firebase/database-node.js:30:641)
    at zg (/home/pi/matrix-os/node_modules/matrix-firebase/node_modules/firebase/database-node.js:216:427)
    at X.h.ec (/home/pi/matrix-os/node_modules/matrix-firebase/node_modules/firebase/database-node.js:151:140)
    at Object.getIDForName (/home/pi/matrix-os/node_modules/matrix-firebase/lib/app.js:186:6)
    at init (/home/pi/matrix-os/lib/service/application.js:158:12)
    at new Application (/home/pi/matrix-os/lib/service/application.js:214:3)

**Comment by brianofrokk3r** [comment]
Hey @nebiljabari I think you're good, we have to remove `START_APP=monitor` from docs. We will do this. Remove it in the meantime and tell me if this works for you.

```
NODE_ENV=rc node index.js
```

**Comment by nebiljabari** [comment]
Ok @brianofrokk3r here is where Iam

- in `~/matrix-os` if I do : `NODE_ENV=rc node index.js`
The LED light spin then stop. In the console some output no error

- in `~/matrix-os` if I do : `NODE_ENV=dev node index.js`
The LED light spin endlessly. In the console bunch of output + each second : `sdk <3-> +1s {}`
when I kill some minor error about child process

- in `~/matrix-os/apps` if I do : `NODE_ENV=rc node index.js` (same result for =dev)
No LED light. In the console :
```
Matrix OS Application Library Loading...
                               ..
               ..
                           ..
C :                ..          ..  ..
```
- in `~/matrix-os/apps/clock.matrix` if I do : `NODE_ENV=rc node index.js` (same result for =dev)
```
Matrix OS Application Library Loading...
clock.matrix is not the same as config name: monitor
    ..
                  ..
                           ..
C :     ..            ..       ..
```

**Comment by brianofrokk3r** [comment]
LED spin means stack initialization, so that's good. Stay on `rc`. In your app, view the `config.yaml` and make sure the folder name matches the `name` property. Currently adding a ticket to make sure this field is updated automagically when an application is created. 

If you see:
```
 _  _ ____ ___ ____ _ _  _ 
 |\/| |__|  |  |__/ |  \/   [o__o]
 |  | |  |  |  |  \ | _/\_  v0.5.0
```

You've instantiated.

**Comment by nebiljabari** [comment]
@eighteyes just saw your comment update

> START_APP=clock node index.js should start the clock

- in `~/matrix-os` it make the spin light then stop same as `node index.js` + the error wrote previously except istead : `monitor` it's logically wrote `clock`

- in `~/matrix-os/apps` or `~/matrix-os/apps/clock` same as upper "dots and dots"

**Comment by nebiljabari** [comment]
Yes @brianofrokk3r I have this output

```
Installed Apps: 
Local Apps: [ 'clock.matrix' ]
Local / Installed Δ 1
MXSS Connected: http://rc-mxss.admobilize.com
=<[^\/^]>= [my-device-id] ready
 _  _ ____ ___ ____ _ _  _ 
 |\/| |__|  |  |__/ |  \/   [o__o]
 |  | |  |  |  |  \ | _/\_  v0.5.0
```

Then I kill the process because I can't enter any command (bad mouv ?). I even tried to let it run and in another terminal window launching the clock. Same dots result

**Comment by brianofrokk3r** [comment]
No, all good! You have to run applications from the CLI interface now. You can run the system as a background process like so if you want:

```
nohup node index.js > mos.out 2>&1 &
```

From there you can tail the response:

```
tail -F mos.out
```

We'll be putting this as a service in the next few days so you won't have to do that soon.

**Comment by nebiljabari** [comment]
> You have to run applications from the CLI interface now

I don't clearly understand. Do you mean that : `sudo matrix start <app-name>` ?
Here is my full "unworking" procedure :
- 1 - `sudo matrix set env rc` // when my rPi boot (just to be sure)
```
Logged Out Successfully
Env: rc
```
- 2 - `sudo matrix login`
```
Login -- username <my-mail>
Login -- password 
 Login Successful : <my-mail>
```
- 3 - `sudo matrix use <my-device-id>`
`Now using device id: <my-device-id>`

- 4 - `source .envrc`

- 5 - `malos` // in another terminal window
```
**************
MALOS starting
**************

You can query specific driver info using port 20012.
Registered driver IMU with port 20013.
Registered driver Humidity with port 20017.
Registered driver Everloop with port 20021.
Registered driver Pressure with port 20025.
Registered driver UV with port 20029.
```
- 6 - `cd matrix-os`
- 7 - `nohup node index.js > mos.out 2>&1 &` // The lights spin 3 times

`[1] 1306`
- 8 - in folder `matrix-os` or `matrix-os/apps` or `matrix-os/apps/clock.matrix` (same result)

`sudo matrix start clock.matrix`
```
Starting:  clock.matrix <my-device-id>
  Started:  clock.matrix
pi@raspberrypi:~/matrix-os $ // Nothing happened
```

ps: if I `tail -F mos.out`: Error lines from my 4th comment
`Error: No user installed firebase application: clock.matrix` etc... 
+
`Cannot start another instance of : clock.matrix`

ps2: 
> In your app, view the config.yaml

We have : `name: clock` but even if I put : `name: clock.matrix` same thing

Ok @brianofrokk3r & @eighteyes I will stop here and wait your coming improvements. Good Luck

**Comment by nebiljabari** [comment]
Ok @eighteyes maybe a good clue I think you've set the default branch on dev and not master
when I `git clone https://github.com/matrix-io/matrix-os.git` 
I have to specified `-b master `otherwise I get the dev branch

Same thing if I go to your repo from the web : https://github.com/matrix-io/matrix-os
Branch: dev (by default)

**Comment by nebiljabari** [comment]
solved here :[ issue #11 ](https://github.com/matrix-io/matrix-os/issues/11#issuecomment-251808291)
